From b2989ba89800a11fb96c9915a7ae8604f50cef46 Mon Sep 17 00:00:00 2001
From: Tony Camuso <tcamuso@redhat.com>
Date: Mon, 29 Feb 2016 10:24:01 -0500
Subject: Some changes were made to avoid stack overflow:

Cherry-picked from:

commit ac33b39dfd317a9c55a86124e08319410d096b32
Author: Ning Sun <ning.sun@intel.com>
Date:   Sat Aug 1 08:25:32 2015 -0700

    Some changes were made to avoid stack overflow:

    1) Increased BSP and AP stacks to 8K and 2K separately in boot.S
    2) Changed a local variable to static variable in memlog_write function of printk.c

    Signed-off-by Safayet Ahmed (GE Global Research) <Safayet.Ahmed@ge.com>

    Signed-off-by Ning Sun <ning.sun@intel.com>
---
 tboot/common/boot.S   |    4 ++--
 tboot/common/printk.c |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/tboot/common/boot.S b/tboot/common/boot.S
index 0400a78..6beeb59 100644
--- a/tboot/common/boot.S
+++ b/tboot/common/boot.S
@@ -39,8 +39,8 @@
 #include <page.h>
 #include <processor.h>
 
-#define BSP_STACK_SIZE		4096
-#define AP_STACK_SIZE		1024
+#define BSP_STACK_SIZE		0x2000
+#define AP_STACK_SIZE		0x1000
 
 #define cs_sel      1<<3
 #define ds_sel      2<<3
diff --git a/tboot/common/printk.c b/tboot/common/printk.c
index a141475..3ca4f55 100644
--- a/tboot/common/printk.c
+++ b/tboot/common/printk.c
@@ -84,7 +84,7 @@ static void memlog_init(void)
 static void memlog_write( const char *str, unsigned int count)
 {
     /* allocate a 32K temp buffer for compressed log  */
-    char buf[32*1024];
+    static char buf[32*1024];
     char *out=buf;
 
     if ( g_log == NULL || count > g_log->max_size )
-- 
1.7.1

